docs/sql_permissions.md

```diff
@@ -0,0 +1,82 @@
+# SQL Permissions
+
+Promscale adopts a Role Based Access Control (RBAC) philosophy to managing permissions
+at the SQL layer. Many users will choose to simply use a single PostgreSQL user for all
+Promscale operations (using a single owner user as defined belows).
+But we also allow users more granular control over permissions management.
+
+Promscale defines several roles:
+- **Owner** this is the owner of the Promscale schema and objects. It is defined
+  as the user that originally installed/migrated Promscale. This user needs to be
+  able to install several PostgreSQL extensions and thus either needs to be a
+  PostgreSQL superuser or use [pgextwlist](https://github.com/dimitri/pgextwlist) to authorize
+  a non-superuser to install the extensions(see below). The owner should be consistent
+  every time you migrate the Promscale schema.
+- **prom_reader** this role is allowed to read but not modify the data stored by Promscale.
+- **prom_writer** is allowed to read Promscale data as well as write (insert) new data. This
+  role is not allowed to modify or delete data. Includes all permissions of prom_reader.
+- **prom_modifier** is able to read, write, and modify all Promscale data. Includes all permissions of prom_writer.
+- **prom_admin** is allowed to change the configuration options associated with Promscale. This
+  includes data retention policies, chunk intervals, etc. Includes all permissions of prom_writer.
+- **prom_maintenance** is allowed to execute maintenance tasks such as compression and data retention jobs.
+  mostly only used externally unless using a CRON job to execute_maintenance() instead of the jobs framework.
+  Includes all permissions of prom_reader.
+
+To run Promscale without schema migration you need to use a PostgreSQL user that has the `prom_modifier` role
```

Comment:
This is confusingly worded, partially because of lack of symmetry with the following line. Perhaps a table with the users running as starting with owner and what they can and cannot do would be useful? 

ie 
```
user , schema_migration, delete_series_api, write_api, read_api
owner   x.                            x                                   x             x
maint                                   x                                   x.            x
```

etc.?

Reply:
well that didn't work because whitespace, but yeah, you get the idea